src/libraries/System.Net.Http/src/System/Net/Http/SocketsHttpHandler/ConnectionPool/HttpConnectionPool.cs

```diff
@@ -561,31 +562,33 @@ public async ValueTask<HttpResponseMessage> SendWithVersionDetectionAndRetryAsyn
             }
         }
 
-        private async ValueTask<(Stream, TransportContext?, IPEndPoint?)> ConnectAsync(HttpRequestMessage request, bool async, CancellationToken cancellationToken)
+        private async ValueTask<(Stream, TransportContext?, Activity?, IPEndPoint?)> ConnectAsync(HttpRequestMessage request, bool async, CancellationToken cancellationToken)
         {
             Stream? stream = null;
             IPEndPoint? remoteEndPoint = null;
+            Activity? activity = CreateActivity()?.Start();
```

Comment:
let's make sure we don't forget to pick relevant attributes for this one and hopefully TLS, DNS and connect spans.